apps/saleor-app-checkout/backend/payments/providers/adyen/utils.ts

```diff
@@ -1,32 +1,43 @@
 import {
-  MoneyFragment,
   OrderFragment,
   TransactionActionEnum,
   TransactionEventFragment,
+  TransactionFragment,
+  TransactionStatus,
+  TransactionUpdateInput,
 } from "@/saleor-app-checkout/graphql";
 import { Types } from "@adyen/api-library";
+import currency from "currency.js";
+import { getTransactionAmountGetterAsMoney } from "../../utils";
+import { failedEvents } from "./consts";
 
 export const getAdyenAmountFromSaleor = (float: number) => parseInt((float * 100).toFixed(0), 10);
 export const getSaleorAmountFromAdyen = (integer: number) => parseFloat((integer / 100).toFixed(2));
 
+const OperationsEnum = Types.notification.NotificationRequestItem.OperationsEnum;
+const EventCodeEnum = Types.notification.NotificationRequestItem.EventCodeEnum;
+
 export const mapAvailableActions = (
-  operations: Types.notification.NotificationRequestItem.OperationsEnum[]
-): TransactionActionEnum[] =>
-  operations.map((operation) => {
-    if (operation === Types.notification.NotificationRequestItem.OperationsEnum.Capture) {
+  operations: Types.notification.NotificationRequestItem.OperationsEnum[] | undefined
+): TransactionActionEnum[] => {
+  if (!operations) return [];
+
+  return operations.map((operation) => {
+    if (operation === OperationsEnum.Capture) {
       return "CHARGE";
     }
 
-    if (operation === Types.notification.NotificationRequestItem.OperationsEnum.Refund) {
+    if (operation === OperationsEnum.Refund) {
       return "REFUND";
     }
 
-    if (operation === Types.notification.NotificationRequestItem.OperationsEnum.Cancel) {
+    if (operation === OperationsEnum.Cancel) {
       return "VOID";
     }
 
     throw "OperationsEnum out of bounds";
```

Comment:
Since you're already refactoring this, we could us a `switch` or an object to make it more concise. Like:

```ts
switch (operation) {
  case OperationsEnum.Capture:
    return "CHARGE";
  case OperationsEnum.Refund:
    return "REFUND";
  case OperationsEnum.Cancel:
    return "VOID";
  default:
    throw new Error("OperationsEnum out of bounds");
}
```